packages/lune/src/config.ts

```diff
@@ -15,11 +21,16 @@ export interface LuneCfg {
    * @default false
    */
   minify?: boolean;
-
-  /** esbuild plugins that run before Lune's transforms */
-  prePlugins?: Plugin[];
-  /** esbuild plugins that run after Lune's transforms */
-  postPlugins?: Plugin[];
+  esbuild?: {
```

Comment:
I would much prefer that, instead of a breaking change to the config, we had something more along the lines of
```ts
type LuneCfg = {
    minify?: boolean;
  }
  & ({
        builder: "esbuild";
        prePlugins?: Plugin[];
        postPlugins?: Plugin[];
     }
   | {
        builder: "rolldown";
        input: InputOptions;
        output: OutputOptions;
      });
```

Reply:
sounds good